optuna/trial/_frozen.py

```diff
@@ -321,8 +322,15 @@ def _validate(self) -> None:
                     "`datetime_complete` is supposed to be None for an unfinished trial."
                 )
 
+        if self.state == TrialState.PRUNED and self._values is not None:
+            raise ValueError("`value` is supposed not to be set for a pruned trial.")
+        if self.state == TrialState.FAIL and self._values is not None:
+            raise ValueError("`value` is supposed not to be set for a failed trial.")
         if self.state == TrialState.COMPLETE and self._values is None:
             raise ValueError("`value` is supposed to be set for a complete trial.")
```

Comment:
```suggestion
            raise ValueError("values should be set for a complete trial.")
```
[nit] Just making the format the same.